R/dv8.R

```diff
@@ -0,0 +1,228 @@
+  depends_files_src <- unique(list(depends_table$src)[[1]])
+  depends_files_dest <- unique(list(depends_table$dest)[[1]])
+  depends_files <- c(depends_files_src, depends_files_dest)
+
+  # Get unique file names
+  depends_files <- unique(depends_files)
+
+  # Sort the file names
+  variables <- sort(depends_files, method="radix")
+
+  # Make indices for the filenames
+  variables_indices <- 1:length(variables)
+  # Make named list with filenames as keys and indices as values
+  names(variables_indices) <- variables
+
+  # List to hold cells' indices
+  cells_indices <- 1:nrow(depends_table)
+
+
+  # get the cell from each row in the depends table
+  getCell <- function(i){
+    depends_src <- depends_table[["src"]][[i]]
+    depends_dest <- depends_table[["dest"]][[i]]
+
+    # Look at first value in parse_depends table, get matching index in vars array
+    src_index <- variables_indices[depends_src] - 1
+    dest_index <- variables_indices[depends_dest] - 1
+
+    values = list()
+    parameters = list("Call", "Import", "Use", "Parameter", "Contain", "Create", "Extend",
+                      "Return","Implement", "Cast", "Throw", "Annotation")
+
+    # Get all the parameter values for a specific cell
+    for (j in 1: length(parameters)) {
+      current_param <- depends_table[i][[parameters[[j]]]]
+      if (current_param > 0){
+        values[[parameters[[j]]]] <- current_param
+      }
+    }
+
+    return(list(src=src_index, dest=dest_index, values=data.frame(values)))
+  }
+
+  # Get the cells
+  cells <- lapply(cells_indices, getCell)
+  cells_df <- data.table::data.table(jsonlite::fromJSON(jsonlite::toJSON(cells, auto_unbox = TRUE)))
+
+  # Sort by the src_index and dest_index if is_sorted is true
+  if (is_sorted==TRUE){
+    data.table::setorder(cells_df, cols = "src", "dest")
+  }
+
+  sdsm_json <- list(schemaVersion="1.0", name=paste0(project_name, "-sdsm"), variables=variables, cells=cells_df)
+
+  json_df <- jsonlite::fromJSON(jsonlite::toJSON(sdsm_json, auto_unbox = TRUE))
+
+  # Unbox each of the cell values (should be values: object, not values: [object])
+  json_df$cells$values <- lapply(json_df$cells$values, jsonlite::unbox)
+
+  # Save the json to a file
+  jsonlite::write_json(json_df, paste0("../rawdata/dv8/", project_name, ".json"), auto_unbox=TRUE)
+}
+
+#' Convert gitlog table to a hdsm.json
+#'
+#' Converts a gitlog table into an *-hdsm.json binary.
+#'
+#' @param gitlog_table parsed gitlog table
+#' @param project_name project name for the hdsm.json
+#' @param is_sorted whether the json is sorted by src and dest
+#' @export
+#' @family dv8
+gitlog_to_hdsmj <- function(gitlog_table, project_name, is_sorted=FALSE){
+  # Call preliminary functions to get graph and cochange for the files
+  gitlog_graph <- transform_gitlog_to_bipartite_network(gitlog_table, mode ="commit-file")
+  cochange_table <- bipartite_graph_projection(gitlog_graph, mode = FALSE, is_intermediate_projection = FALSE)
```

Comment:
Were you able to check the values of co-change here match to DV8?

Reply:
Also curious if this is the source of the slow down on the function brought up. Could you try running line by line to see where is the performance bottleneck? 

Same ask for the dependencies_sdsmj(). Where is the bottleneck?

Reply:
Hi Carlos, the values of co-change did not exactly match, likely since my output file has a different number of variables than the dv8 output file https://github.com/sailuh/kaiaulu/issues/184.

>Also curious if this is the source of the slow down on the function brought up. Could you try running line by line to see where is the performance bottleneck?

I don't believe transform_gitlog_to_bipartite_network() nor bipartite_graph_projection() were the bottleneck, running these lines individually.

Well, actually the table generated by bipartite_graph_projection() may be the problem. I got a huge table when I ran the original Apache Helix repository on my gitlog_to_hdsmj() function. I don't remember how large it was, but the json it generated crashed my RStudio for a while, leaving me with a white screen whenever I tried to open the app. Had to manually move/delete a file to fix that.

>Same ask for the dependencies_sdsmj(). Where is the bottleneck?

I don't think there's a bottleneck for this function. The only one I noticed being slow was gitlog_to_hdsmj().